### DISCUSSION TITLE
Valetudo 2023.12.0

### DISCUSSION THREAD

**Discussion by Hypfer** [Hypfer/Valetudo#1945] [Release Discussions]
<div align="center">
    <img src="https://raw.githubusercontent.com/Hypfer/Valetudo/master/assets/logo/valetudo_logo_with_name.svg" width="800" alt="valetudo">
    <p align="center"><h2>2023.12.0</h2></p>
</div>

Quality of Life, Maintenance, Merry Christmas and it's finally 37c3 time

## Map Action Icons

All buttons floating on top of map views now have icons!<br/>
It's one of these changes that leaves you wondering why you didn't do that earlier. Thanks again for the input, @alufers!

![image](https://github.com/Hypfer/Valetudo/assets/974410/8b95d6c7-c353-4951-b2be-98baa31ecad9)

With the play icon there, maybe less newcomers will click on the large play button provided by the BasicControlCapability?

![image](https://github.com/Hypfer/Valetudo/assets/974410/1fdb65ba-15d3-43c1-9cb8-1283ebf69601)

I don't know what I expected for a join icon.

![image](https://github.com/Hypfer/Valetudo/assets/974410/b67581a4-6465-4b61-9caf-14a72b1d2a51)

Virtual restrictions not only received custom icons but also a slight overhaul, adding counters, shortening button labels and adding a "Clear" button for consistency.

## Home Assistant device classes

If you're using Home Assistant, with this release, you will see the battery state of the robot as an icon in the device overview.
@mill1000 noticed that we previously forgot to set the correct device class on the battery entity.

![image](https://github.com/Hypfer/Valetudo/assets/974410/8f3af7a0-1c42-40b6-8c42-47a8a036d39e)

Neat!

## Maintenance Maintenance Maintenance

This release bumps (almost) all dependencies to their latest versions including the necessary refactorings caused by breaking API changes.
Boring, tedious work that just needs to be done periodically.
On top of that, the NodeJS base binary was upgraded to v20 from the v18 used previously.

I still need to upgrade from react-router v5 to v6, but that will be a new year's resolution.

With these changes, from a user perspective, things should be exactly the same as before.

## Fixes to bugs and annoyances

- Disabling edge mopping via the Dreame Quirk now properly reports that it actually was disabled
- SSE subscriptions now have a delay until they disconnect from the server to avoid unnecessary dis- and reconnects when changing views
- The error message thrown on invalid characters in the Wi-Fi password now tries to inform the user that Valetudo is not doing that just to ruin their day - Honestly it's sad that this change was required

## 37c3

![image](https://github.com/Hypfer/Valetudo/assets/974410/5db57895-7329-4928-b711-75d3dd565fff)

With that whole pandemic thingy finally being resolved and behind us, the Chaos Communication Congress will return to Hamburg!
Dennis and I will be there with rooting PCBs for Dreame, Valetudo stickers, a weird robot that may or may not be rootable and also [a new talk by Dennis](https://fahrplan.events.ccc.de/congress/2023/fahrplan/events/11943.html).<br/>
Come say hi :)

If you don't know how to do that because you have no idea how I look like, just stand in front of a candle-lit mirror and say "multi-floor support" 13 times. Looking for a table with some vacuum robots on it might also work.

## The usual

If you want to see Valetudo on more robots and/or like this release, you might want to consider donating:

[https://github.com/sponsors/Hypfer](https://github.com/sponsors/Hypfer)

[https://builder.dontvacuum.me/donations.txt](https://builder.dontvacuum.me/donations.txt)

### Features

- **ui**: Replace new icons with custom new icons + remove new icon dependency [`5621418`](http://github.com/Hypfer/Valetudo/commit/56214184f74a22c61826af9b2a58337b4c29285b)
- **ui**: Add some nice icons to all map action buttons [`4e958da`](http://github.com/Hypfer/Valetudo/commit/4e958da68b810153fcf39a17bff63bc1d78d14dd)
- **vendor.roborock**: Dock dustbin consumable for all auto-empty docks [`e3d13fa`](http://github.com/Hypfer/Valetudo/commit/e3d13fa521b9944e9e2040f96847002f5feaa1e9)
- Set the embedded process priority to below normal for good measure [`eadc2a7`](http://github.com/Hypfer/Valetudo/commit/eadc2a7a047531722c1a59729fcb5cde5e565bfa)
- **mqtt**: Add device_class attribute to battery and WiFi sensors for Home Assistant  (#1910) [`aac0f51`](http://github.com/Hypfer/Valetudo/commit/aac0f51da1d8009cee8380dcee81dda8eafdf3f1)
- **vendor.dreame**: Edge mopping for the L10 Ultra [`744fe20`](http://github.com/Hypfer/Valetudo/commit/744fe20833deec24a069029a22df8df6a0465d9c)

### Fixes

- **miio**: Error message clarifications [`d17b299`](http://github.com/Hypfer/Valetudo/commit/d17b299839a5169753bc7a207aea37ef36c58b54)
- **ui**: Fix achievement display and keep SSE connections alive for additional 500ms after the last subscriber disconnected [`c20731d`](http://github.com/Hypfer/Valetudo/commit/c20731d86102b8a7b83def05c3b6d03f2b3fdf6f)
- **vendor.dreame**: Remove CollisionAvoidantNavigationControl from the x10+ since it is not supported by the firmware [`d4f37e4`](http://github.com/Hypfer/Valetudo/commit/d4f37e4ea518a6d2ddc457db89a73d9255dba159)
- **vendor.dreame**: Fix edge mopping quirk state reporting [`9e4617b`](http://github.com/Hypfer/Valetudo/commit/9e4617ba3589ff1e069a99237f07826a674291ed)

### Refactoring

- Minor code cleanup [`ddb49e7`](http://github.com/Hypfer/Valetudo/commit/ddb49e7112d8422ae9505e77b103c83ba924f793)
- Bump all dependencies to latest versions + minor code cleanup [`5a364ef`](http://github.com/Hypfer/Valetudo/commit/5a364efdeacae0dd1ce907bd16317e264501b58a)

### Chores

- **release**: 2023.12.0 [`ce683c5`](http://github.com/Hypfer/Valetudo/commit/ce683c52f018bb07f0fb23f874031d0832aa7f01)
- Bump dependencies [`d63e325`](http://github.com/Hypfer/Valetudo/commit/d63e32529c7b4c8411bb225f43ecc14d139b7574)
- **build**: Bump to NodeJS v20.10.0 [`25eaa76`](http://github.com/Hypfer/Valetudo/commit/25eaa76499d990f6d9b40898443c618117903b6e)
- Disallow some more footguns with eslint [`9d9213c`](http://github.com/Hypfer/Valetudo/commit/9d9213c84e366ed91447c73bb0834d08ecd1be08)
- **vendor.dreame**: Add test case with a ginormous map file from an industrial building [`0d5a7df`](http://github.com/Hypfer/Valetudo/commit/0d5a7df48b61b12b9216f3818a4f3271a9ec0a0a)
- **build**: Bump to NodeJS v18.18.2 [`ba90e9b`](http://github.com/Hypfer/Valetudo/commit/ba90e9b6601832f1f65b2993303f99f4e854b85f)

<hr /><em>This discussion was created from the release <a href='https://github.com/Hypfer/Valetudo/releases/tag/2023.12.0'>Valetudo 2023.12.0</a>.</em>

**Comment by benbender** [comment]
Thanks for your hard work! See you @ 37c3!
